### PR TITLE
pfsync status report

### PR DESCRIPTION
**Pull request by kprovost**
pfsync status report

### CODE REVIEW

File: 2018q4/pfsync.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+was loaded.
+
+This was improved by splitting the pfsync queues into buckets, based on the
+state id. This ensures that updates for a given connection always end up
+in the same bucket, which allows pfsync to still collapse multiple
+updates into one, while allowing multiple cores to proceed at the same
+time.
+The buckets are independently locked, allowing multiple cores to proceed at once.
+
+The number of buckets is tunable, but defaults to 2 x number of cpus.
```
Comment (wblock) [inline 2018q4/pfsync.md]: s/2 x/twice the/

File: 2018q4/pfsync.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Contact: Kristof Provost <kp@freebsd.org>
+Link:	 None
+
+While pf itself can operate on multiple states simultaneously
+(on different cores), pfsync could not. It used a single PFSYNC_LOCK.
+This greatly reduced throughput on multicore systems as soon as pfsync
+was loaded.
+
+This was improved by splitting the pfsync queues into buckets, based on the
+state id. This ensures that updates for a given connection always end up
```
Comment (wblock) [inline 2018q4/pfsync.md]: s/id/ID/

File: 2018q4/pfsync.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+in the same bucket, which allows pfsync to still collapse multiple
```
Comment (wblock) [inline 2018q4/pfsync.md]: s/which allows/allowing/

File: 2018q4/pfsync.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+updates into one, while allowing multiple cores to proceed at the same
+time.
+The buckets are independently locked, allowing multiple cores to proceed at once.
+
+The number of buckets is tunable, but defaults to 2 x number of cpus.
+Benchmarking has shown improvement, depending on hardware and setup, from ~30%
```
Comment (wblock) [inline 2018q4/pfsync.md]: Flip around for clarity and to remove the aside:
`` Benchmarking has shown improvement of 30 to 100% depending on hardware and setup. ``

**Comment by kprovost** [comment]
I believe I've addressed all of the remarks. Thanks for the review.
I'm not sure if there's some magic button here to move forward, or if I've already done everything required.

**Comment by trasz** [comment]
Hello.  Warren, is there anything more to do before this can be merged?

**Comment by trasz** [comment]
Thanks for submission; merged.

**Pull request merged**